src/firstboot/firstboot.c

```diff
@@ -458,8 +458,20 @@ static int process_locale(int rfd) {
         return 1;
 }
 
+static bool keymap_exists_bool(const char *name) {
+        return keymap_exists(name) > 0;
+}
+
+static bool keymap_is_ok(int rfd, const char *name) {
+        assert(rfd >= 0);
+
+        return dir_fd_is_root(rfd) ? keymap_exists_bool(name) : keymap_is_valid(name);
+}
+
 static int prompt_keymap(int rfd) {
         _cleanup_strv_free_ char **kmaps = NULL;
+        bool (*is_valid)(const char *name) = dir_fd_is_root(rfd) ? keymap_exists_bool
```

Comment:
dir_fd_is_root() possibly an error here. which you treat as "true" here. Which you probably shouldnt.

I mean, it's fine to treat failures as true or as false sometimes, dependening on the case, but this should always be made explicit via `!= 0` or `> 0` checks, to underline you are aware this function doesn't just return a boolean.

Reply:
(that said, we nowadays prefer if there's at least debug logging about unexpected errors. i.e. a quick log_debug_errno() line if the check returns < 0. similar for the keymap_exists() check above)